From 6b6f281e4fc3b64f158a049234e43a46119c652e Mon Sep 17 00:00:00 2001
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Mon, 29 Mar 2010 16:14:47 -0300
Subject: [PATCH 12/23] Monitor: Convert do_memory_save() to cmd_new_ret()

RH-Author: Luiz Capitulino <lcapitulino@redhat.com>
Message-id: <1269879298-30711-13-git-send-email-lcapitulino@redhat.com>
Patchwork-id: 8177
O-Subject: [PATCH 12/23] Monitor: Convert do_memory_save() to cmd_new_ret()
Bugzilla: 563491
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
(cherry picked from commit 9869622ee8d613d5d3cbb2a7d47a6192ecfe2e2b)
---
 monitor.c       |    9 +++++++--
 qemu-monitor.hx |    2 +-
 2 files changed, 8 insertions(+), 3 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 monitor.c       |    9 +++++++--
 qemu-monitor.hx |    2 +-
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/monitor.c b/monitor.c
index 06caf03..dd82705 100644
--- a/monitor.c
+++ b/monitor.c
@@ -1465,7 +1465,7 @@ static void do_print(Monitor *mon, const QDict *qdict)
     monitor_printf(mon, "\n");
 }
 
-static void do_memory_save(Monitor *mon, const QDict *qdict, QObject **ret_data)
+static int do_memory_save(Monitor *mon, const QDict *qdict, QObject **ret_data)
 {
     FILE *f;
     uint32_t size = qdict_get_int(qdict, "size");
@@ -1474,13 +1474,14 @@ static void do_memory_save(Monitor *mon, const QDict *qdict, QObject **ret_data)
     uint32_t l;
     CPUState *env;
     uint8_t buf[1024];
+    int ret = -1;
 
     env = mon_get_cpu();
 
     f = fopen(filename, "wb");
     if (!f) {
         qemu_error_new(QERR_OPEN_FILE_FAILED, filename);
-        return;
+        return -1;
     }
     while (size != 0) {
         l = sizeof(buf);
@@ -1494,8 +1495,12 @@ static void do_memory_save(Monitor *mon, const QDict *qdict, QObject **ret_data)
         addr += l;
         size -= l;
     }
+
+    ret = 0;
+
 exit:
     fclose(f);
+    return ret;
 }
 
 static void do_physical_memory_save(Monitor *mon, const QDict *qdict,
diff --git a/qemu-monitor.hx b/qemu-monitor.hx
index 0680fa5..3044aa6 100644
--- a/qemu-monitor.hx
+++ b/qemu-monitor.hx
@@ -670,7 +670,7 @@ ETEXI
         .params     = "addr size file",
         .help       = "save to disk virtual memory dump starting at 'addr' of size 'size'",
         .user_print = monitor_user_noop,
-        .mhandler.cmd_new = do_memory_save,
+        .cmd_new_ret = do_memory_save,
     },
 
 STEXI
-- 
1.7.0.3

